include/RAJA/util/View.hpp

```diff
@@ -185,7 +187,13 @@ struct MultiView
         data(rhs.data)
   {}
 
-  RAJA_INLINE void set_data(pointer_type data_ptr) { data = data_ptr; }
+  RAJA_HOST_DEVICE RAJA_INLINE void set_layout(layout_type const& ly) { layout = ly; }
+
+  RAJA_HOST_DEVICE RAJA_INLINE void set_data(pointer_type data_ptr) { data = data_ptr; }
```

Comment:
You can add an explicit cast to support const data types.
```suggestion
  RAJA_HOST_DEVICE RAJA_INLINE void set_data(pointer_type data_ptr) { data = nc_pointer_type(data_ptr); }
```